File: wpilibcExamples/src/main/cpp/examples/GamepadSelectable/cpp/opmode/ScoreAuto.cpp
// Copyright (c) FIRST and other WPILib contributors.
// Open Source Software; you can modify and/or share it under the terms of
// the WPILib BSD license file in the root directory of this project.

#include "opmode/ScoreAuto.hpp"

#include <string>

#include "wpi/driverstation/DriverStationDisplay.hpp"

ScoreAuto::ScoreAuto() {
  wpi::DriverStationDisplay::SetMode(wpi::DriverStationDisplay::Mode::Line);
  m_selectable.AddOptions("Target", {"High", "Mid", "Low"});
  m_selectable.AddIntegerOptions("Delay", 0, 5, 1);
  m_selectable.AddDoubleOptions("Speed", 0.25, 1.0, 0.25);
}

void ScoreAuto::DisabledPeriodic() {
  m_selectable.Update();
  wpi::DriverStationDisplay::UpdateLines();
}

void ScoreAuto::Start() {
  m_target = m_selectable.GetSelected("Target");
  m_delay = m_selectable.GetSelectedInteger("Delay");
  m_speed = m_selectable.GetSelectedDouble("Speed");
}

void ScoreAuto::Periodic() {
  wpi::DriverStationDisplay::AddData("Selected Auto", "Score");
  wpi::DriverStationDisplay::AddData("Target", m_target);
  wpi::DriverStationDisplay::AddData("Delay", std::to_string(m_delay));
  wpi::DriverStationDisplay::AddData("Speed", std::to_string(m_speed));
  wpi::DriverStationDisplay::UpdateLines();
}

File: wpilibjExamples/src/main/java/org/wpilib/examples/dsgamepadchooser/ScoreAuto.java
// Copyright (c) FIRST and other WPILib contributors.
// Open Source Software; you can modify and/or share it under the terms of
// the WPILib BSD license file in the root directory of this project.

package org.wpilib.examples.dsgamepadchooser;

import java.util.List;
import org.wpilib.driverstation.DriverStationDisplay;
import org.wpilib.driverstation.DriverStationDisplay.Mode;
import org.wpilib.driverstation.DSGamepadChooser;
import org.wpilib.opmode.Autonomous;
import org.wpilib.opmode.PeriodicOpMode;

/** An autonomous opmode with chooser scoring parameters. */
@Autonomous(name = "Score Auto")
public class ScoreAuto extends PeriodicOpMode {
  private final DSGamepadChooser chooser = new DSGamepadChooser(0);
  private String target = "High";
  private int delay;
  private double speed = 0.25;

  /** Creates a Score Auto opmode. */
  public ScoreAuto() {
    DriverStationDisplay.setMode(Mode.Line);
    chooser.addOptions("Target", List.of("High", "Mid", "Low"));
    chooser.addIntegerOptions("Delay", 0, 5, 1);
    chooser.addDoubleOptions("Speed", 0.25, 1.0, 0.25);
  }

  /** Called periodically while this opmode is selected and the robot is disabled. */
  @Override
  public void disabledPeriodic() {
    chooser.update();
    DriverStationDisplay.updateLines();
  }

  /** Called once when the robot transitions from disabled to enabled. */
  @Override
  public void start() {
    target = chooser.getSelected("Target");
    delay = chooser.getSelectedInteger("Delay");
    speed = chooser.getSelectedDouble("Speed");
  }

  /** Called periodically while this opmode is enabled. */
  @Override
  public void periodic() {
    DriverStationDisplay.addData("Selected Auto", "Score");
    DriverStationDisplay.addData("Target", target);
    DriverStationDisplay.addData("Delay", delay);
    DriverStationDisplay.addData("Speed", speed);
    DriverStationDisplay.updateLines();
  }
}
